File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        ls=[]
        if len(s)<=1:
            return False
        dc={")":"(","]":"[","}":"{"}
        for item in s:
            if item in dc.keys():
                if ls and ls[-1]==dc[item]:
                    ls.pop() 
                else:
                    ls.append(item)               
            else:
                ls.append(item)
        if ls == []:
            return True
        else:
            return False 
        
        

File: 20-valid-parentheses/valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = []
    let n = s.length

    let matching = {")":"(","]":"[","}":"{"}
    for (let i=0;i<n;i++){

        if (s[i] in matching){
            if ((stack) && (stack.pop()==matching[s[i]])){
                continue
            }
            else{
                return false
            }
        }
        else{
            stack.push(s[i])
        }

    }

    if (stack.length==0){
        return true
    }
    return false
};